src/CheckInput.java

```diff
@@ -0,0 +1,199 @@
+                if (pin.length() != 4) {
+                    System.out.println(RED_BOLD + "Not a valid PIN. 4 numerical digits required." + ANSI_RESET);
+                    return false;
+                }
+                    return true;
+            } catch (Exception e) {
+                System.out.println(RED_BOLD + "Not a valid PIN. 4 numerical digits required." + ANSI_RESET);
+                return false;
+            }
+    }
+
+    public boolean isValidLogInCode(String option){
+        try{
+            int code = Integer.parseInt(option);
+            if(code <1 || code >3)
+            {
+                System.out.println(RED_BOLD + "Not a valid code." + ANSI_RESET);
+                return false;
+            }else {
+                return true;
+            }
+
+        }catch(Exception e)
+        {
+            System.out.println(RED_BOLD + "Not a valid code." + ANSI_RESET);
+            return false;
+        }
+
+    }
+
+    public boolean isValidMenuCode(String option){
+        try{
+            int code = Integer.parseInt(option);
+            if(code <1 || code >8)
+            {
+                System.out.println(RED_BOLD + "Not a valid code." + ANSI_RESET);
+                return false;
+            }else {
+                return true;
+            }
+
+        }catch(Exception e)
+        {
+            System.out.println(RED_BOLD + "Not a valid code." + ANSI_RESET);
+            return false;
+        }
+
+    }
+
+    public boolean isValidLongInput(String input){
+        try {
+            long amount = Long.parseLong(input);
+            if(amount<=0)
+            {
+                System.out.println(RED_BOLD + "Not a valid amount." + ANSI_RESET);
+                return false;
+            }
+            return true;
+        }catch(Exception e){
+            System.out.println(RED_BOLD + "Not a valid amount." + ANSI_RESET);
+            return false;
+        }
+    }
+
+    public boolean isValidDoubleInput(String input){
+        try {
+            double amount = Double.parseDouble(input);
+            if(amount<=0)
+            {
+                System.out.println(RED_BOLD + "Not a valid amount." + ANSI_RESET);
+                return false;
+            }
+            return true;
+        }catch(Exception e){
+            System.out.println(RED_BOLD + "Not a valid amount." + ANSI_RESET);
+            return false;
+        }
+    }
+
+    public boolean isValidProviderCode(String option){
+        try{
+            int code = Integer.parseInt(option);
+            if(code <1 || code >4)
+            {
+                System.out.println(RED_BOLD + "Not a valid provider code." + ANSI_RESET);
+                return false;
+            }else {
+                return true;
+            }
+
+        }catch(Exception e)
+        {
+            System.out.println(RED_BOLD + "Not a valid provider code." + ANSI_RESET);
+            return false;
+        }
+
+    }
+
+    public String validAmount(){
```

Comment:
refactor this to validate the input you get as parameter, not asking it in the method
public String validAmount(string input){}

Reply:
I moved the method entirely from CheckInput because it was not really a checking method, and I moved it in providers, where I am calling the isValidDouble from the CheckInput.
And I also fixed an issue of the payBills method. :) it was not returning the last up-to-date currentBalance.